### PR TITLE
FINERACT-1060: Do not allow delete / update of system user

### PR DESCRIPTION
**Pull request by ptuomola**
FINERACT-1060

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Commit message starts with the issue number from https://issues.apache.org/jira/projects/FINERACT/. Ex: FINERACT-646 Pockets API.

- [x] Coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions have been followed.

- [x] API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm has been updated with details of any API changes.

- [x] Integration tests have been created/updated for verifying the changes made.

- [x] All Integrations tests are passing with the new commits.

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the list.)

Our guidelines for code reviews is at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide

### CODE REVIEW

**Comment by vorburger** [comment]
Test Failure just due to FINERACT-1016.

**Review by awasum — APPROVED** [review]
LGTM. @vorburger  merge this before you go?

I am just trying to catch up on all this issues and may not have the proper contexts...

File: fineract-provider/src/main/java/org/apache/fineract/useradministration/domain/AppUser.java [hunk 1]
```diff
@@ -264,6 +264,12 @@ public void updateRoles(final Set<Role> allRoles) {
 
         final String usernameParamName = "username";
         if (command.isChangeInStringParameterNamed(usernameParamName, this.username)) {
+
+            // TODO Remove this check once we are identifying system user based on user ID
```
Comment (awasum) [inline fineract-provider/src/main/java/org/apache/fineract/useradministration/domain/AppUser.java:268]: Do we have an issue explaining how to implement this todo or just to log the issue on Jira so someone can pick it up later?

Comment (ptuomola) [reply fineract-provider/src/main/java/org/apache/fineract/useradministration/domain/AppUser.java:268]: I've created a new issue for this - FINERACT-1068

**Pull request merged**